src/main/java/build/buildfarm/instance/AbstractServerInstance.java

```diff
@@ -325,19 +333,22 @@ protected void validateAction(Action action) {
     // invalid action?
     filesUniqueAndSortedPrecondition(action.getOutputFilesList());
     filesUniqueAndSortedPrecondition(action.getOutputDirectoriesList());
-    if (commandDigest.getSizeBytes() != 0 || Command.getDefaultInstance().toByteString().size() != 0) {
-      Command command;
-      try {
-        command = Command.parseFrom(getBlob(commandDigest));
-      } catch (InvalidProtocolBufferException ex) {
-        Preconditions.checkState(
-            false,
-            INVALID_DIGEST);
-        return;
-      }
-      environmentVariablesUniqueAndSortedPrecondition(
-          command.getEnvironmentVariablesList());
+
+    Preconditions.checkState(commandDigest.getSizeBytes() != 0, INVALID_DIGEST);
```

Comment:
A bit redundant, this would be handled by line 350 as well.

Reply:
Agreed, was thinking too much in a null pointer mindset.